pkg/cache/v3/cache_test.go

```diff
@@ -78,27 +79,27 @@ func TestHeartbeatResponseGetDiscoveryResponse(t *testing.T) {
 	}
 
 	discoveryResponse, err := resp.GetDiscoveryResponse()
-	assert.Nil(t, err)
-	assert.Equal(t, discoveryResponse.VersionInfo, resp.Version)
-	assert.Equal(t, len(discoveryResponse.Resources), 1)
-	assert.False(t, isTTLResource(discoveryResponse.Resources[0]))
+	require.NoError(t, err)
+	assert.Equal(t, discoveryResponse.GetVersionInfo(), resp.Version)
+	assert.Len(t, discoveryResponse.GetResources(), 1)
```

Comment:
```suggestion
	require.Len(t, discoveryResponse.GetResources(), 1)
```

We should require here because the next assertion depends on it.